solutions/devsprint-michael-douglas-2/DeliveryApp/Screens/Components/LoadingView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  LoadingView.swift
+//  DeliveryApp
+//
+//  Created by Hyago Henrique on 21/02/22.
+//
+
+import UIKit
+
+final class LoadingView: UIView {
+    
+    private let isTesting: Bool
+    
+    // MARK: - View Components
+    
+    private lazy var loadingContainer: UIView = {
+        let container = UIView(frame: .zero)
+        container.translatesAutoresizingMaskIntoConstraints = false
+        container.isOpaque = true
+        container.isHidden = false
+        container.backgroundColor = .white
+        return container
+    }()
+    
+    private lazy var loadingSpinner: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView(frame: .zero)
+        indicator.translatesAutoresizingMaskIntoConstraints = false
+        indicator.style = .large
+        indicator.isHidden = false
+        indicator.color = .black
+        return indicator
+    }()
+    
+    private lazy var loadingMessage: UILabel = {
+        let label = UILabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = "Carregando..."
+        label.isHidden = false
+        label.textAlignment = .center
+        label.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        label.numberOfLines = 0
+        label.textColor = .black
+        return label
+    }()
+    
+    // MARK: - Super Methods
+    
+    init(isTesting: Bool = false) {
+        self.isTesting = isTesting
+        super.init(frame: .zero)
+        
+        self.setupViews()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Internal Methods
+    func setLoadingMessage(_ message: String?) {
+        loadingMessage.text = message ?? "Carregando..."
+    }
+    
+    func updateLoading(_ isLoading: Bool) {
+        isHidden = !isLoading
+        if !isTesting {
+            if isLoading {
+                loadingSpinner.startAnimating()
+            } else {
+                loadingSpinner.stopAnimating()
+            }
+        }
+    }
```

Comment:
Ao invés de fazer um `if`, que tal utilizar um `guard`?? Assim, seu código fica mais legível e limpo.

```suggestion
    func updateLoading(_ isLoading: Bool) {
        isHidden = !isLoading
        
        guard shouldAnimate else { return }
        
        if isLoading {
             loadingSpinner.startAnimating()
        } else {
             loadingSpinner.stopAnimating()
        }
    }
```

Reply:
Show, realmente bem melhor!